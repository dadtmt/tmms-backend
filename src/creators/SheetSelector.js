import React, { PropTypes } from 'react'
import { ControlLabel, FormControl, FormGroup } from 'react-bootstrap'

import CharacterSheet from './CharacterSheet'

const SheetSelector = ({ selectSheet, selectedSheet, sheets }) => <div>
  <FormGroup controlId='selectCharacter'>
    <ControlLabel>Select Character</ControlLabel>
    <FormControl
      componentClass='select'
      onChange={selectSheet}
      placeholder='select'
      value={selectedSheet.id}
    >
      <option key='new' value=''>create a new character</option>
      {sheets.map(
        ({ node: { id, name } }) =>
          <option key={id} value={id} >{name}</option>
      )}
    </FormControl>
  </FormGroup>
  <CharacterSheet initialValues={selectedSheet} />
</div>

SheetSelector.propTypes = {
  selectSheet: PropTypes.func.isRequired,
  selectedSheet: PropTypes.object.isRequired,
  sheets: PropTypes.array.isRequired
}

export default SheetSelector
