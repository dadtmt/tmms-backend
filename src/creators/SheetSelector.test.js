import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo'

import SheetSelector from './SheetSelector'

/*eslint-disable*/
const sheets = [
  {
    'node': {
      'id': 'id_1',
      'charateristics': null,
      'description': 'sdfsdf',
      'image': 'dsfdf',
      'inventory': null,
      'isPlayer': true,
      'name': 'Jojo le first',
      '__typename': 'Sheets'
    },
    '__typename': 'SheetsEdge'
  },
  {
    'node': {
      'id': 'id_2',
      'charateristics': null,
      'description': null,
      'image': null,
      'inventory': null,
      'isPlayer': false,
      'name': 'toto',
      '__typename': 'Sheets'
    },
    '__typename': 'SheetsEdge'
  }
]

/*eslint-enable*/

const props = {
  selectSheet: jest.fn(),
  selectedSheet: { name: 'new character' },
  sheets
}

const client = new ApolloClient()

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <ApolloProvider client={client} >
      <SheetSelector {...props} />
    </ApolloProvider>,
    div
  )
})

it('renders correctly', () => {
  expect(
    renderer.create(<ApolloProvider client={client}>
      <SheetSelector {...props} />
    </ApolloProvider>).toJSON()
  ).toMatchSnapshot()
})
