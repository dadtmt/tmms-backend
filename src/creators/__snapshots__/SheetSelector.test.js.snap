exports[`test renders correctly 1`] = `
<div>
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor="selectCharacter">
      Select Character
    </label>
    <select
      className="form-control"
      id="selectCharacter"
      onChange={[Function]}
      placeholder="select"
      type={undefined}
      value={undefined}>
      <option
        value="">
        create a new character
      </option>
      <option
        value="id_1">
        Jojo le first
      </option>
      <option
        value="id_2">
        toto
      </option>
    </select>
  </div>
  <form
    onSubmit={[Function]}>
    <div
      className="checkbox"
      style={undefined}>
      <label>
        <input
          checked=""
          disabled={false}
          name="isPlayer"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="checkbox"
          value="" />
        player sheet
      </label>
    </div>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor="name">
        Character name:
      </label>
      <input
        className="form-control"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder={undefined}
        type="text"
        value="" />
      <span
        className="help-block" />
    </div>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor="description">
        description:
      </label>
      <input
        className="form-control"
        id="description"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder={undefined}
        type="text"
        value="" />
      <span
        className="help-block" />
    </div>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor={undefined}>
        image:
      </label>
      <input
        className="form-control"
        id={undefined}
        name="image"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Image url"
        type="text"
        value="" />
      <img
        className="img-rounded"
        src="" />
    </div>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-heading">
        Characteristics (add and edit)
      </div>
      <div
        className="panel-body">
        <div>
          <div
            className="form-group">
            <button
              className="btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button">
              <span
                className="glyphicon glyphicon-plus" />
            </button>
          </div>
          <ul
            className="list-group" />
        </div>
      </div>
    </div>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-heading">
        Gear (add and edit item)
      </div>
      <div
        className="panel-body">
        <div>
          <div
            className="form-group">
            <button
              className="btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button">
              <span
                className="glyphicon glyphicon-plus" />
            </button>
          </div>
          <ul
            className="list-group" />
        </div>
      </div>
    </div>
    <div
      className="form-group">
      <button
        className="btn btn-default"
        disabled={true}
        title="save"
        type="submit">
        <span
          className="glyphicon glyphicon-save" />
      </button>
    </div>
  </form>
</div>
`;
